File: src/TMS-DotNet02-Online.Shchypakin.FitnessApp.Data/Enities/Membership.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;


namespace TMS_DotNet02_Online.Shchypakin.FitnessApp.Data.Enities
{
    public class Membership
    {
        public int Id { get; set; }

        public int ClientId { get; set; }

        public Client Client { get; set; }

        public DateTime Start { get; set; }

        public DateTime End { get; set; }

        public int MembershipTypeId { get; set; }

        public MembershipType MembershipType { get; set; }

        public int MembershipSizeId { get; set; }

        public MembershipSize MembershipSize { get; set; }

        public bool IsActive { get; set; }

        public bool Online { get; set; }

        public ICollection<MembershipHistoryRecord> MembershipHistoryRecords { get; set; }
    }
}

File: src/client/src/app/_models/membership.ts
import { MembershipHistoryRecord } from "./membershipHistoryRecord";
import { MembershipSize } from "./membershipSize";
import { MembershipType } from "./membershipType";


export interface Membership {
    id: number;
    start: Date;
    end: Date;
    membershipType: MembershipType;
    membershipSize: MembershipSize;
    visitsLeft: number;
    isActive: boolean;
    membershipHistoryRecords: MembershipHistoryRecord[];
}
